utils/writeable/src/lib.rs

```diff
@@ -319,6 +319,57 @@ pub trait Writeable {
     }
 }
 
+/// A [`Writeable`] that is either one type or another type.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[allow(clippy::exhaustive_enums)] // defined to be an enumeration over two types
+pub enum EitherWriteable<W0, W1> {
```

Comment:
`either::Either` is the standard Rust type to do this. Can we use that instead of defining our own?

Reply:
Fewer deps seems nice... we already have EitherYoke and EitherProvider defined in their respective crates

@Manishearth opinion?

Reply:
I'm okay with an optional dep here. I'm also somewhat open to moving EitherYoke to Either if possible.

> a "standard"

"standard Rust type" does not necessarily mean it is "a standard", these have two different implications. I'd say `either` _is_ the standard Rust type for this.

Reply:
In general when it comes to deps we already pull in in a couple places I'm less worried about them. I still want our utils to be simple but it's fine if it's optional deps for crates that are extremely commonly found in Rust.